MmSupervisorPkg/Drivers/MmPeiLaunchers/MmIplPei.c

```diff
@@ -656,63 +794,14 @@ ExecuteMmCoreFromMmram (
   // if Loading module at Fixed Address feature is enabled, the SMM core driver will be loaded to
   // the address assigned by build tool.
   //
-  if (PcdGet64 (PcdLoadModuleAtFixAddressEnable) != 0) {
-    ASSERT (FALSE);
-    // //
-    // // Get the fixed loading address assigned by Build tool
-    // //
-    // Status = GetPeCoffImageFixLoadingAssignedAddress (&ImageContext);
-    // if (!EFI_ERROR (Status)) {
-    //   //
-    //   // Since the memory range to load SMM CORE will be cut out in SMM core, so no need to allocate and free this range
-    //   //
-    //   PageCount = 0;
-    //   //
-    //   // Reserved Mmram Region for SmmCore is not used, and remove it from MmramRangeCount.
-    //   //
-    //   gMmCorePrivate->MmramRangeCount --;
-    // } else {
-    //   DEBUG ((DEBUG_INFO, "LOADING MODULE FIXED ERROR: Loading module at fixed address at address failed\n"));
-    //   //
-    //   // Allocate memory for the image being loaded from the EFI_SRAM_DESCRIPTOR
-    //   // specified by MmramRange
-    //   //
-    //   PageCount = (UINTN)EFI_SIZE_TO_PAGES((UINTN)ImageContext.ImageSize + ImageContext.SectionAlignment);
-
-    //   ASSERT ((MmramRange->PhysicalSize & EFI_PAGE_MASK) == 0);
-    //   ASSERT (MmramRange->PhysicalSize > EFI_PAGES_TO_SIZE (PageCount));
-
-    //   MmramRange->PhysicalSize -= EFI_PAGES_TO_SIZE (PageCount);
-    //   MmramRangeSmmCore->CpuStart = MmramRange->CpuStart + MmramRange->PhysicalSize;
-    //   MmramRangeSmmCore->PhysicalStart = MmramRange->PhysicalStart + MmramRange->PhysicalSize;
-    //   MmramRangeSmmCore->RegionState = MmramRange->RegionState | EFI_ALLOCATED;
-    //   MmramRangeSmmCore->PhysicalSize = EFI_PAGES_TO_SIZE (PageCount);
-
-    //   //
-    //   // Align buffer on section boundary
-    //   //
-    //   ImageContext.ImageAddress = MmramRangeSmmCore->CpuStart;
-    // }
-  } else {
-    //
-    // Allocate memory for the image being loaded from the EFI_SRAM_DESCRIPTOR
-    // specified by MmramRange
-    //
-    PageCount = (UINTN)EFI_SIZE_TO_PAGES ((UINTN)ImageContext.ImageSize + ImageContext.SectionAlignment);
-
-    ASSERT ((MmramRange->PhysicalSize & EFI_PAGE_MASK) == 0);
-    ASSERT (MmramRange->PhysicalSize > EFI_PAGES_TO_SIZE (PageCount));
-
-    MmramRange->PhysicalSize        -= EFI_PAGES_TO_SIZE (PageCount);
-    MmramRangeSmmCore->CpuStart      = MmramRange->CpuStart + MmramRange->PhysicalSize;
-    MmramRangeSmmCore->PhysicalStart = MmramRange->PhysicalStart + MmramRange->PhysicalSize;
-    MmramRangeSmmCore->RegionState   = MmramRange->RegionState | EFI_ALLOCATED;
-    MmramRangeSmmCore->PhysicalSize  = EFI_PAGES_TO_SIZE (PageCount);
+  //
+  // Allocate memory for the image being loaded from unallocated mmram range
+  //
+  PageCount = (UINTN)EFI_SIZE_TO_PAGES ((UINTN)ImageContext.ImageSize + ImageContext.SectionAlignment);
 
-    //

```

Comment:
The logic for finding the ImageAddress is duplicated with this new function and in the entry point (lines 1291-1336).  The code there should probably be deleted so we don't do duplicate work.